tracer/contrib/redigo/tracedredis.go

```diff
@@ -0,0 +1,101 @@
+package tracedredis
+
+import (
+	"context"
+	"github.com/DataDog/dd-trace-go/tracer"
+	"github.com/DataDog/dd-trace-go/tracer/ext"
+	redis "github.com/garyburd/redigo/redis"
+	"net"
+	"net/url"
+	"strconv"
+	"strings"
+)
+
+type TracedConn struct {
+	redis.Conn
+	tracer  *tracer.Tracer
+	service string
+	network string
+	host    string
+	port    string
+}
+
+func TracedDial(service string, tracer *tracer.Tracer, network, address string, options ...redis.DialOption) (TracedConn, error) {
+	c, err := redis.Dial(network, address)
+	addr := strings.Split(address, ":")
+	host := addr[0]
+	port := addr[1]
+	tracer.SetServiceInfo(service, "redis", ext.AppTypeDB)
+	tc := TracedConn{c, tracer, service, network, host, port}
+	return tc, err
+}
+
+func (tc *TracedConn) SetService(service string) {
+	tc.tracer.SetServiceInfo(service, "redis", ext.AppTypeDB)
+	tc.service = service
+}
+
+func TracedDialURL(service string, tracer *tracer.Tracer, rawurl string, options ...redis.DialOption) (TracedConn, error) {
+	u, err := url.Parse(rawurl)
+	if err != nil {
+		return TracedConn{}, err
+	}
+	host, port, err := net.SplitHostPort(u.Host)
+	if err != nil {
+		host = u.Host
+		port = "6379"
```

Comment:
We set port to 6379 only when can't get the port via net.SplitHostPort (this is the err that gets in the if) https://github.com/garyburd/redigo/blob/master/redis/conn.go#L226